src/controllers/dlgcontrollerlearning.cpp

```diff
@@ -206,9 +205,46 @@ void DlgControllerLearning::slotChooseControlPressed() {
 void DlgControllerLearning::startListening() {
     // Start listening as soon as we're on this page -- that way advanced
     // users don't have to specifically click the "Learn" button.
-    // Get the underlying type of the Controller. This will call
-    // one of the visit() methods below immediately.
-    m_pController->accept(this);
+    // Disconnect everything in both directions so we don't end up with duplicate connections
+    // after pressing the "Learn Another" button
+    MidiController* pMidiController = static_cast<MidiController*>(m_pController);
```

Comment:
Use qobject_cast for this purpose. A static_cast does not do any verification.

Reply:
fixed